avocado/utils/multipath.py

```diff
@@ -0,0 +1,188 @@
+"""
+
+import time
+from . import distro
+from . import process
+
+
+class MultipathError(Exception):
+    """
+    Multipath Error
+    """
+
+    def __init__(self, msg):
+        self.msg = msg
+
+    def __str__(self):
+        return self.msg
+
+
+def form_conf_mpath_file(blacklist="", defaults_extra=""):
+    """
+    Form a multipath configuration file, and restart multipath service.
+
+    :parm blacklist: Entry in conf file to indicate blacklist section.
+    :parm defaults_extra: Extra entry in conf file in defaults section.
+    """
+    mpath_fp = open("/etc/multipath.conf", "w")
+    mpath_fp.write("defaults {\n")
+    mpath_fp.write("    find_multipaths yes\n")
+    mpath_fp.write("    user_friendly_names yes\n")
+    if defaults_extra:
+        mpath_fp.write("    %s\n" % defaults_extra)
+    mpath_fp.write("}\n")
+    if blacklist:
+        mpath_fp.write("blacklist {\n")
+        mpath_fp.write("    %s\n" % blacklist)
+        mpath_fp.write("}\n")
+    mpath_fp.close()
+    cmd = "cat /etc/multipath.conf"
+    process.system(cmd, ignore_status=True)
+    multipath_service(operation='restart')
+
+
+def multipath_service(operation='restart'):
+    """
+    Start, Stop, Restart multipath service.
+
+    :parm operation: multipath service operation to perform.
+                     Can be one of start, stop, restart.
+
+    :raise: :class:`MultipathError`, if ``multipath is inactive``.
+    """
+    if distro.detect().name == 'Ubuntu':
+        service_name = "multipath-tools"
+    else:
+        service_name = "multipathd"
+    cmd = "service %s %s" % (service_name, operation)
+    time.sleep(5)
+    process.system(cmd, ignore_status=True)
+    if 'start' in operation and not multipath_isactive():
+        cmd = "cat /etc/multipath.conf"
+        process.system(cmd, ignore_status=True)
+        raise MultipathError("Multipath service not active")
+
+
+def multipath_isactive():
+    """
+    Checks if multipath service is active or not.
+
+    :return: True if service is active, False if inactive.
+    """
+    if distro.detect().name == 'Ubuntu':
+        service_name = "multipath-tools"
+    else:
+        service_name = "multipathd"
+    cmd = "service %s status" % service_name
+    out = process.system_output(cmd, ignore_status=True)
+    for line in out.splitlines():
+        if 'Active' in line:
+            return line.split()[1] == 'active'
+    return False
+
+
+def path_exists(path):
+    """
+    Checks if a given path exists.
+
+    :return: True if path exists, False if does not exist.
+    """
+    cmd = "multipath -l %s" % path
+    if process.system(cmd, ignore_status=True) != 0:
+        return False
+    return True
+
+
+def get_mpath_name(wwid):
+    """
+    Get multipath name for a given wwid.
+
+    :parm wwid: wwid of multipath device.
+
+    :return: Name of multipath device.
+    """
+    cmd = "multipath -l %s" % wwid
+    return process.system_output(cmd, ignore_status=True).split()[0]
```

Comment:
Do you have a special reason to ignore status? In case of error I'd prefer a better message than `IndexError: list index out of range`...

Also would you please add `sudo=True` everywhere where necessary? (eg. this command) It does nothing when running as root, but it prepends sudo when running as an ordinary user.